app/mm-redux/client/client4.ts

```diff
@@ -2839,6 +2839,13 @@ export default class Client4 {
         );
     };
 
+    getGroups = async (filterAllowReference = false) => {
+        return this.doFetch(
+            `${this.getBaseRoute()}/groups${buildQueryString({filter_allow_reference: filterAllowReference})}`,
```

Comment:
so somewhere here you'd need to handle `since`

Reply:
and make sure when you call the client with since you dont call the reducer if there is no new data

Reply:
@enahum @migbot 
Is it okay for me to add the websocket reconnect stuff in a separate PR as I would like this PR to focus on getting the main functionality merged in or would you guys prefer if I got it done with this PR (it would require me to open a server PR). Personally it would just be less confusing for myself to do it separately but I would like to get your opinion.

Reply:
I am okay with the reconnect logic being done in a separate PR.

Reply:
Having out of date groups showing up in your autocomplete list isnt bad especially since the server would validate and not send mentions when not needed anyway

Reply:
Ok! I have created two tickets for this. One to implement the server side and one to implement the mobile side.
https://mattermost.atlassian.net/browse/MM-24692
https://mattermost.atlassian.net/browse/MM-24693